src/core/meshcop/joiner_router.cpp

```diff
@@ -0,0 +1,364 @@
+/*
+ *  Copyright (c) 2016, Nest Labs, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements the Joiner Router role.
+ */
+
+#include <assert.h>
+#include <stdio.h>
+
+#include <common/code_utils.hpp>
+#include <common/encoding.hpp>
+#include <common/logging.hpp>
+#include <meshcop/joiner_router.hpp>
+#include <thread/meshcop_tlvs.hpp>
+#include <thread/thread_netif.hpp>
+#include <thread/thread_uris.hpp>
+
+using Thread::Encoding::BigEndian::HostSwap16;
+
+namespace Thread {
+namespace MeshCoP {
+
+JoinerRouter::JoinerRouter(ThreadNetif &aNetif):
+    mRelayTransmit(OPENTHREAD_URI_RELAY_TX, &HandleRelayTransmit, this),
+    mNetif(aNetif)
+{
+    mNetif.GetCoapServer().AddResource(mRelayTransmit);
+}
+
+ThreadError JoinerRouter::Start(void)
+{
```

Comment:
This is a TODO, but Joiner Router role should start whenever there is an Active Commissioner.